timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -558,54 +559,51 @@ def _train_multi(
 
         logger.info(f"Models that will be trained: {list(m.name for m in models)}")
 
-        time_limit_model_split = time_limit
-        if time_limit is not None and len(models) > 0:
-            time_limit_model_split /= len(models)
+        if split_time_limit is None:
+            split_time_limit = hyperparameter_tune_kwargs is not None
 
+        num_models = len(models)
+        if num_models > 1:  # ensemble is only fit len(models) > 1
+            num_models += int(self.enable_ensemble)
         model_names_trained = []
         for i, model in enumerate(models):
-            if hyperparameter_tune_kwargs is not None:
-                time_left = time_limit_model_split
+            if time_limit is None:
+                time_left = None
+            else:
+                time_left = time_limit - (time.time() - time_start)
+                if time_left <= 0:
+                    logger.info(f"Stopping training due to lack of time remaining. Time left: {time_left:.2f} seconds")
+                    break
+
+            if time_left is not None and split_time_limit:
+                time_left_for_model = time_left / (num_models - i)
```

Comment:
so this is a different behavior still than what I would expect from the docstring. you're not giving each model equal shots from the beginning, but deciding on the fly which model gets what. However in principle, if models obey time limits strictly, then there should be no regressions I guess.

Reply:
Some models (e.g., Naive or SeasonalNaive) may complete training before the time limit is exceeded. With this logic we can redistribute the extra time among the remaining models.